### PR TITLE
Make label requirements for merging configurable

### PR DESCRIPTION
**Pull request by kargakis**
We want to block merging PRs in Origin on more labels than the existing required set. This is not a problem with `tide` but `tide` is far from ready for prime time today.

/cc @cjwagner @apelisse 
@stevekuznetsov @jupierce @eparis fyi

/area mungegithub

Signed-off-by: Michail Kargakis <mkargaki@redhat.com>

### CODE REVIEW

File: mungegithub/mungers/submit-queue_test.go [hunk 1]
```diff
@@ -845,6 +846,21 @@ func TestSubmitQueue(t *testing.T) {
...
+			commits:          Commits(), // Modified at time.Unix(7), 8, and 9
+			ciStatus:         SuccessStatus(),
+			lastBuildNumber:  LastBuildNumber(),
+			gcsResult:        SuccessGCS(),
+			retest1Pass:      true,
+			retest2Pass:      true,
+			reason:           noAdditionalLabelMessage("kind/blocker"),
+			state:            "pending",
+		},
+		{
```
Comment (cjwagner) [inline mungegithub/mungers/submit-queue_test.go]: Please add a test case for the scenario in which a PR has the required additional labels.

**Comment by kargakis** [comment]
> This looks good. Should we get rid of GateApproved and make the approved label an AdditionalRequiredLabel?

We could possibly do it for all required labels. I want to consider it in a follow-up though.

File: mungegithub/mungers/submit-queue_test.go (hunk 1 above)
Comment (kargakis) [reply mungegithub/mungers/submit-queue_test.go]: good call, added one

**Comment by apelisse** [comment]
Should we move existing required flags to that parameters?

**Comment by cjwagner** [comment]
> Should we move existing required flags to that parameters?

I think so. @kargakis wants that to be a separate PR though which makes sense.

/lgtm

**Pull request merged**